### PR TITLE
feat: adds `InMemoryProvider`

### PR DESCRIPTION
**Pull request by maxveldink**
## This PR

- Adds `InMemoryProvider` that can be used for initializing with flags in process, as defined in [Appendix A](https://openfeature.dev/specification/appendix-a#in-memory-provider).
- Adds `debug` gem for development and testing
- Enables the RSpec focus feature

### How to test
You can instantiate and use an InMemoryProvider in an OF project and see that flags can be set through it.

### CODE REVIEW

**Review by beeme1mr — APPROVED** [review]

**Comment by maxveldink** [comment]
@open-feature/sdk-ruby-maintainers Going to merge this in tomorrow morning if there are no issues with it

**Review by josecolella — APPROVED** [review]

**Pull request merged**
